src/coinsselectionalgorithm.cpp

```diff
@@ -0,0 +1,596 @@
+        #endif
+        size_t tempTotalSize = 0;
+        CAmount tempTotalAmount = 0;
+        unsigned int tempTotalSelection = 0;
+        int windowFrontIndex = maxIndex;
+        int windowBackIndex = maxIndex;
+        bool admissibleFound = false;
+        bool bestAdmissibleFound = false; // "best" for this specific algorithm implementation
+        for (; !stopRequested && windowFrontIndex >= 0; --windowFrontIndex)   
+        {
+            #if COINS_SELECTION_ALGORITHM_PROFILING
+            ++iterations;
+            #endif
+
+            // insert new coin in selection
+            tempSelection[windowFrontIndex] = 1;
+            tempTotalSize += sizes[windowFrontIndex];
+            tempTotalAmount += amounts[windowFrontIndex];
+            tempTotalSelection += 1;
+
+            // check upper-limit constraints
+            while (tempTotalSize > availableTotalSize ||
+                   tempTotalAmount > targetAmountPlusOffset)
+            {
+                #if COINS_SELECTION_ALGORITHM_PROFILING
+                ++iterations;
+                #endif
+       
+                // if admissible solution still not found, pop from back of the sliding window
+                if (!admissibleFound)
+                {
+                    tempSelection[windowBackIndex] = 0;
+                    tempTotalSize -= sizes[windowBackIndex];
+                    tempTotalAmount -= amounts[windowBackIndex];
+                    tempTotalSelection -= 1;
+                    --windowBackIndex;
+                }
+                // if admissible solution already found, pop from front of the sliding window only the element just inserted (and break)
+                else
+                {
+                    tempSelection[windowFrontIndex] = 0;
+                    tempTotalSize -= sizes[windowFrontIndex];
+                    tempTotalAmount -= amounts[windowFrontIndex];
+                    tempTotalSelection -= 1;
+                    bestAdmissibleFound = true;
+                    break; // almost surely it is useless, but kept for better flow
+                }
+            }
+
+            // check lower-limit constraint
+            if (tempTotalAmount >= targetAmount)
+            {
+                admissibleFound = true;
+                // if best admissible solution already found or reached array end, set optimal solution
+                if (bestAdmissibleFound || windowFrontIndex == 0)
+                {
+                    optimalTotalSize = tempTotalSize;
+                    optimalTotalAmount = tempTotalAmount;
+                    optimalTotalSelection = tempTotalSelection;
+                    for (int index = 0; index < problemDimension; ++index)
+                    {
+                        optimalSelection[index] = tempSelection[index];
+                    }
+                    break;
```

Comment:
As you're doing a copy isn't it better to do this ?
```
optimalSelection.assign(tempSelection.begin(), tempSelection.end()); 
// or as option (less elegant)
// memcpy(optimalSelection.data(), tempSelection.data(), problemDimension); // note I omitted the conversion in bytes as we already know is a vector of bytes
                                                                            // but for the sake of completeness we should multiply for sizeof(elem)
```

Reply:
I agree with your suggestion.
Modified accordingly.